streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java

```diff
@@ -1747,6 +1747,30 @@ public void shouldWrapProcessorsForMaterializedSourceTable() {
         assertThat(counter.numConnectedStateStores(), CoreMatchers.is(1));
     }
 
+    @Test
+    public void shouldWrapProcessorsForToTable() {
```

Comment:
I don't think we need a new test for this, it's covered by `shouldWrapProcessorsForStatelessOperators` right?

If we want to split things up into individual tests then imo we should have a stateless KStream operators and KTable operators -- that is, remove the .toTable().filter().toStream() from shouldWrapProcessorsForStatelessOperators and have two unit tests: this one, plus a new one for `KTable#filter` (once we implement that cc @rodesai )